pallets/capacity/src/lib.rs

```diff
@@ -679,6 +694,43 @@ pub mod pallet {
 			Self::deposit_event(Event::PrecipitatingTokenomicEventSet { at: pte_block_number });
 			Ok(())
 		}
+
+		/// Stakes some amount of tokens to the network and locks it for a certain period for higher rewards and the tokens get unlocked gradually after a certain lock period.
+		/// ### Errors
+		///
+		/// - Error::InvalidTarget if attempting to stake to an invalid target.
+		/// - Error::StakingAmountBelowMinimum if attempting to stake an amount below the minimum amount.
+		/// - Error::CannotChangeStakingType if the staking account exists and staking_type is MaximumCapacity
+		#[pallet::call_index(8)]
+		#[pallet::weight(T::WeightInfo::committed_boost())]
+		pub fn committed_boost(
+			origin: OriginFor<T>,
+			target: MessageSourceId,
+			amount: BalanceOf<T>,
+		) -> DispatchResult {
+			let staker = ensure_signed(origin)?;
+
+			let (mut staking_details, stakable_amount) =
+				Self::ensure_can_stake(&staker, target, amount, StakingType::CommittedBoost)?;
+			staking_details.staking_type = StakingType::CommittedBoost;
```

Comment:
nit: I think this should be done inside `ensure_can_stake` (but I don't feel super strongly about it...)